soh/src/code/z_parameter.c

```diff
@@ -1493,6 +1493,22 @@ void Inventory_SwapAgeEquipment(void) {
             gSaveContext.equips.equipment = gSaveContext.childEquips.equipment;
             gSaveContext.equips.equipment &= 0xFFF0;
             gSaveContext.equips.equipment |= 0x0001;
+        } else if (gSaveContext.n64ddFlag) {
```

Comment:
It sounds like this block should only run when rando AND starting age is adult right? If so add `Randomizer_GetSettingValue(RSK_STARTING_AGE)` the condition here